### PR TITLE
Update to new SDK API

### PR DESCRIPTION
**Pull request by joshua-kim**
## Why this should be merged

See https://github.com/ava-labs/avalanchego/pull/2452

## How this works

Updates to new SDK API

## How this was tested

UTs

### CODE REVIEW

File: peer/network.go [hunk 1]
```diff
@@ -78,9 +78,10 @@ type Network interface {
 	// (length of response divided by request time), and with 0 if the response is invalid.
 	TrackBandwidth(nodeID ids.NodeID, bandwidth float64)
 
-	// NewAppProtocol reserves a protocol identifier and returns a corresponding
-	// client to send messages with
-	NewAppProtocol(protocol uint64, handler p2p.Handler, options ...p2p.ClientOption) (*p2p.Client, error)
+	// NewClient returns a client to send messages with for the given protocol
+	NewClient(protocol uint64, options ...p2p.ClientOption) (*p2p.Client, error)
+	// NewAppProtocol registers a server handler for an application protocol
```
Comment (abi87) [inline peer/network.go:83]: nit: seems outdate comment, drop it

File: peer/network.go [hunk 2]
```diff
@@ -78,9 +78,10 @@ type Network interface {
...
+	AddHandler(protocol uint64, handler p2p.Handler) error
```
Comment (abi87) [inline peer/network.go:84]: q: could you comment on the difference between `AddHandler` and the various `SetGossip/Request/CrossChainRequest/Handler`?

File: peer/network.go (hunk 1 above)
Comment (joshua-kim) [reply peer/network.go:83]: Whoops... removed

File: peer/network.go (hunk 2 above)
Comment (joshua-kim) [reply peer/network.go:84]: `AddHandler` is for the p2p sdk handlers. The existing signatures are for the legacy handlers. Eventually we will delete all the old stuff from this once migration is complete

File: peer/network.go [hunk 3]
```diff
@@ -543,8 +544,12 @@ func (n *network) TrackBandwidth(nodeID ids.NodeID, bandwidth float64) {
...
 }
 
-func (n *network) NewAppProtocol(protocol uint64, handler p2p.Handler, options ...p2p.ClientOption) (*p2p.Client, error) {
-	return n.network.NewAppProtocol(protocol, handler, options...)
+func (n *network) NewClient(protocol uint64, options ...p2p.ClientOption) (*p2p.Client, error) {
+	return n.network.NewClient(protocol, options...)
+}
+
+func (n *network) AddHandler(protocol uint64, handler p2p.Handler) error {
+	return n.network.AddHandler(protocol, handler)
```
Comment (ceyonur) [inline peer/network.go:552]: can we rename the inner `network `struct (p2p)?

Comment (ceyonur) [reply peer/network.go:552]: Or alternatively maybe we can directly embed `*p2p.Network` in to the struct.

File: go.mod [hunk 4]
```diff
@@ -4,7 +4,7 @@ go 1.20
 
 require (
 	github.com/VictoriaMetrics/fastcache v1.10.0
-	github.com/ava-labs/avalanchego v1.10.18-rc.0
+	github.com/ava-labs/avalanchego v1.10.18-0.20231211033727-3c26d85f63c9
```
Comment (ceyonur) [inline go.mod:7]: can we use the same tag for `1.10.123-refactor-client-1`?

File: peer/network.go (hunk 3 above)
Comment (joshua-kim) [reply peer/network.go:552]: Embedding makes this look weird still sine the name is then `n.Network`... i'll just rename this to something else

**Review by abi87 — APPROVED** [review]
lgtm

**Review by darioush — APPROVED** [review]

File: go.mod (hunk 4 above)
Comment (joshua-kim) [reply go.mod]: I'm using a commit hash now since I think the tags don't work with e2e.

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**